File: unittest/scripts/js_devapi/validation/dba_cluster_misconfigurations.js
//@ Initialization
||

//@ create cluster admin
|Cluster admin user 'ca'@'%' created.|

//@ Dba.createCluster (fail because of bad configuration)
||Instance check failed

//@ Dba.createCluster (fail because of bad configuration of parallel-appliers) {VER(>=8.0.23)}
||Instance check failed

//@ Dba.createCluster (succeeds with right configuration of parallel-appliers) {VER(>=8.0.23)}
||

//@ Cluster.addInstance (fail because of bad configuration of parallel-appliers) {VER(>=8.0.23)}
||Instance check failed

//@ Cluster.addInstance (succeeds with right configuration of parallel-appliers) {VER(>=8.0.23)}
||

//@ Setup next test
||

//@ Create cluster fails (one table is not compatible)
|WARNING: The following tables do not have a Primary Key or equivalent column:|
||Instance check failed (RuntimeError)

//@ Enable verbose
||

//@Create cluster fails (one table is not compatible) - verbose mode
|WARNING: The following tables do not have a Primary Key or equivalent column:|
||Instance check failed
//@ Disable verbose
||

//@ Create cluster succeeds (no incompatible table)
||

//@ Dissolve cluster at the end (clean-up)
||

//@ Finalization
||

File: unittest/scripts/py_devapi/validation/dba_cluster_misconfigurations.py
#@ Initialization
||

#@ Dba.createCluster
||

#@ Dissolve cluster (to re-create again)
||

#@ Create cluster fails (one table is not compatible)
||SystemError: RuntimeError: Dba.create_cluster: ERROR: 1 table(s) do not have a Primary Key or Primary Key Equivalent (non-null unique key).

#@ Enable verbose
||

#@<ERR> Create cluster fails (one table is not compatible) - verbose mode
* Checking compliance of existing tables... FAIL
ERROR: 1 table(s) do not have a Primary Key or Primary Key Equivalent (non-null unique key).
	pke_test.t3

Group Replication requires tables to use InnoDB and have a PRIMARY KEY or PRIMARY KEY Equivalent (non-null unique key). Tables that do not follow these requirements will be readable but not updateable when used with Group Replication. If your applications make updates (INSERT, UPDATE or DELETE) to these tables, ensure they use the InnoDB storage engine and have a PRIMARY KEY or PRIMARY KEY Equivalent.


ERROR: Error starting cluster: The operation could not continue due to the following requirements not being met:
Non-compatible tables found in database.
==============================================================================

#@ Disable verbose
||

#@ Create cluster succeeds (no incompatible table)
||

#@ Dissolve cluster at the end (clean-up)
||

#@ Finalization
||
